pkg/sdn/plugin/egress_network_policy.go

```diff
@@ -79,3 +91,57 @@ func (plugin *OsdnNode) UpdateEgressNetworkPolicyVNID(namespace string, oldVnid,
 		plugin.updateEgressNetworkPolicyRules(newVnid)
 	}
 }
+
+func (plugin *OsdnNode) syncEgressDNSPolicyRules() {
```

Comment:
This is really complicated and confusing. You should make the EgressDNS code smarter and this code dumber. egress_network_policy.go shouldn't have to deal with the timers itself... EgressDNS should do that internally, and then only wake up egress_network_policy.go when something has actually changed. So this code would just be something like:

    func (plugin *OsdnNode) syncEgressDNSPolicyRules() {
            for {
                    policyNamespace <- plugin.egressDNS.changed
                    if vnid, err := plugin.policy.GetVNID(policyNamespace); err == nil {
                            plugin.updateEgressNetworkPolicyRules(vnid)
                    } else {
                            glog.Warningf("Could not find netid for namespace %q: %v", policyNamespace, err)
                    }
            }
    }

(Likewise in proxy.go)